tests/pytype_test.py

```diff
@@ -149,6 +149,17 @@ def find_stubs_in_paths(paths: Sequence[str]) -> list[str]:
     return filenames
 
 
+def _get_pkgs_associated_with_requirement(req_name: str) -> list[str]:
+    dist = importlib.metadata.distribution(req_name)
+    toplevel_txt_contents = dist.read_text("top_level.txt")
+    if toplevel_txt_contents is not None:
+        return toplevel_txt_contents.split()
+    if dist.files is None:
+        raise RuntimeError("Can't read find the packages associated with requirement {req_name!r}")
+    maybe_modules = [f.parts[0] if len(f.parts) > 1 else inspect.getmodulename(f) for f in dist.files]
+    return [name for name in maybe_modules if name is not None and "." not in name]
```

Comment:
I drew on the implementation of `packages_distributions()` heavily when writing this function: https://github.com/python/cpython/blob/904aef994262383ae916545908f0578c2d53cf31/Lib/importlib/metadata/__init__.py#L933-L964